pkg/operator/controller/ingress/deployment.go

```diff
@@ -463,6 +466,16 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		}
 	}
 
+	if ci.Spec.Logging != nil && ci.Spec.Logging.Access != nil && ci.Spec.Logging.Access.HTTPCaptureHeaders != nil {
+		env = append(env, corev1.EnvVar{
+			Name:  RouterCaptureHTTPRequestHeaders,
+			Value: serializeCaptureHeaders(ci.Spec.Logging.Access.HTTPCaptureHeaders.Request),
```

Comment:
Is there a guarantee that if `ci.Spec.Logging.Access.HTTPCaptureHeaders` is non-nil, both `ci.Spec.Logging.Access.HTTPCaptureHeaders.Request` and `ci.Spec.Logging.Access.HTTPCaptureHeaders.Response` are also non-nil? What if `ci.Spec.Logging.Access.HTTPCaptureHeaders.Request` were to be nil, but `ci.Spec.Logging.Access.HTTPCaptureHeaders.Response` was non-nil?

Reply:
> Is there a guarantee that if `ci.Spec.Logging.Access.HTTPCaptureHeaders` is non-nil, both `ci.Spec.Logging.Access.HTTPCaptureHeaders.Request` and `ci.Spec.Logging.Access.HTTPCaptureHeaders.Response` are also non-nil?

No.  It is possible to specify `httpCaptureHeaders` with nil `request` and `response` values.  In this case, the operator will specify empty values for `ROUTER_CAPTURE_HTTP_REQUEST_HEADERS` and `ROUTER_CAPTURE_HTTP_RESPONSE_HEADERS`, which is harmless.

> What if `ci.Spec.Logging.Access.HTTPCaptureHeaders.Request` were to be nil, but `ci.Spec.Logging.Access.HTTPCaptureHeaders.Response` was non-nil?

Then `ROUTER_CAPTURE_HTTP_REQUEST_HEADERS` would be empty and `ROUTER_CAPTURE_HTTP_RESPONSE_HEADERS` would be nonempty.  This should be fine; the `range` expressions that https://github.com/openshift/router/pull/139/files adds will iterate over the 0 comma-delimited values and add 0 `capture` stanzas.

Reply:
> the operator will specify empty values for `ROUTER_CAPTURE_HTTP_REQUEST_HEADERS` and `ROUTER_CAPTURE_HTTP_RESPONSE_HEADERS`, which is harmless.

Ah, ok, wasn't sure if empty values were intended or not. Thanks!